core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -156,7 +162,8 @@ private[spark] class AppStatusStore(
     // cheaper for disk stores (avoids deserialization).
     val count = {
       Utils.tryWithResource(
-        if (store.isInstanceOf[InMemoryStore]) {
+        if (isInMemoryStore) {
```

Comment:
After reading the comment in https://github.com/apache/spark/pull/23088/files#diff-3bd1667878f7bda9a56f95e93a80b475R233, I think it is on purpose that only count the "SUCCESS" task when with `InMemoryStore`.
cc @shahidki31

Reply:
Thanks @gengliangwang . History server uses `InMemory` store by default. For Disk store case,  this isn't an optimal way for finding success task. I am yet to raise a PR for supporting for Disk store case.  

I think you just need to add,
 `isInMemoryStore: Boolean =  listener.isDefined || store.isInstanceOf[InMemoryStore]`

Will test your code with all the scenarios.
Any case, this would be temporary as we need to support for Diskstore also.

Reply:
@shahidki31 Thanks for the suggestion.

I am aware that #23088 is to follow the behavior of previous versions of spark.  But I wonder if we can simply show the summary metrics for all the tasks instead of only the "SUCCESS" ones, as all the tasks are listed in the task table. By doing that should also make sense to users. The implementation will be simpler and we don't have to worry about the performance of the disk store.
Also cc @vanzin @srowen

Reply:
I don't know enough to evaluate this one, sorry. The code change itself looks plausible.

Reply:
@gengliangwang If I understand correctly, #23008 is actually fixing this issue. right ?(At least history server case).
Because, `count` is always filtering out the running tasks, as `ExecutorRunTime` will be defined for only finished tasks. But `scan tasks` contains all the tasks, including running tasks. 
https://github.com/apache/spark/blob/a59fdc4b5783be591a236bfc60d1107caa818412/core/src/main/scala/org/apache/spark/status/AppStatusStore.scala#L169

https://github.com/apache/spark/blob/a59fdc4b5783be591a236bfc60d1107caa818412/core/src/main/scala/org/apache/spark/status/AppStatusStore.scala#L263-L265

So, reverting 23008 and hence this PR, would not fix the issue? Please correct me if I am wrong.

Reply:
@shahidki31 Do you mean #23088?
> as ExecutorRunTime will be defined for only finished tasks
That's not true. It is also defined in running tasks as I observed.

> reverting 23008 and hence this PR, would not fix the issue
I am not trying to revert anything. Just out of curiosity for the feature.

Reply:
I guess the underlying question you guys are asking is: should the metrics count only successful tasks, or all non-failed tasks?

I don't remember the behavior in 2.2 (which is what #23088 was trying to emulate?), which should have the correct answer.

Reply:
@vanzin In Spark-2.2, as per this line, it doesn't count metrics of a running task. https://github.com/apache/spark/blob/7c7d7f6a878b02ece881266ee538f3e1443aa8c1/core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala#L340

verified locally also, 
 `sc.parallelize(1 to 160, 1).map( i => Thread.sleep(1000)).collect()`

![Screenshot 2019-08-09 at 4 03 09 AM](https://user-images.githubusercontent.com/23054875/62742208-4beb1a80-ba5b-11e9-97a7-b01f9515b5fd.png)

Reply:
So sounds like the current behavior is consistent with how Spark has behaved in the past, and this change is proposing a different approach.

A quick look at the history shows the current behavior has been that in forever...

Also, that pointed me at another thing that can be seen in the screenshots. The table header says "Summary Metrics for X *Completed* Tasks". So this change would be making that wrong...

Reply:
Ok, I need to backtrack here a little. My bad, I think I know why I'm confused.

I've been starting from the PR title, and it confused me a little bit. Could you change it so it describes what the change is doing? e.g. "Ignore non-succeeded tasks when calculating metric summaries."

Let me re-review trying to ignore that and focusing just on the code.

Reply:
@vanzin Got it. I have updated the title.
The PR itself is about mismatching the kvstore, and leading to the wrong cache for task summary.

Reply:
May be we need to add a test in the AppStatusStoreSuite? There all the stores are tested against InMemory store only, I think.

Reply:
@shahidki31 sure, I have added test cases.